docs/source/online_portfolio_selection/pattern_matching.rst

```diff
@@ -0,0 +1,581 @@
+
+    # CORN
+    # Compute Correlation Driven Nonparametric Learning with no given weights, window of 1, and rho of 0.3.
+    corn = CORN(window=1, rho=0.3)
+    corn.allocate(asset_prices=stock_prices, resample_by='W', verbose=True)
+
+    # Compute Correlation Driven Nonparametric Learning with user given weights, window of 3, and rho of 0.5.
+    corn1 = CORN(window=3, rho=0.5)
+    corn1.allocate(asset_prices=stock_prices, weights=some_weight)
+
+    # CORN-U
+    # Compute Correlation Driven Nonparametric Learning - Uniform with no given weights, window range of 10, and rho of 0.3.
+    cornu = CORNU(window=10, rho=0.3)
+    cornu.allocate(asset_prices=stock_prices, resample_by='W', verbose=True)
+
+    # Compute Correlation Driven Nonparametric Learning - Uniform with user given weights, window range of 5, and rho of 0.1.
+    cornu1 = CORNU(window=5, rho=0.1)
+    cornu1.allocate(asset_prices=stock_prices, weights=some_weight)
+
+    # CORN-K
+    # Compute Correlation Driven Nonparametric Learning - K with no given weights, window range of 10, rho of 7, and k of 2.
+    cornk = CORNK(window=10, rho=7, k=2)
+    cornk.allocate(asset_prices=stock_prices, resample_by='W', verbose=True)
+
+    # Compute Correlation Driven Nonparametric Learning - K with user given weights, window range of 5, rho of 3, and k of 1.
+    cornk1 = CORNK(window=5, rho=3, k=1)
+    cornk1.allocate(asset_prices=stock_prices, weights=some_weight)
+
+    # Recalculate k for cornk1 to save computational time of generating all experts.
+    cornk1.recalculate_k(k=3)
+
+    # Get the latest predicted weights.
+    corn.weights
+
+    # Get all weights for the strategy.
+    cornk.all_weights
+
+    # Get portfolio returns.
+    cornu.portfolio_return
+
+    # Get each object of the generated experts.
+    cornk1.experts
+
+    # Get each experts parameters.
+    cornu.expert_params
+
+    # Get all expert's portfolio returns over time.
+    cornu1.expert_portfolio_returns
+
+    # Get capital allocation weights.
+    cornk1.weights_on_experts
+
+----
+
+Symmetric Correlation Driven Nonparametric Learning
+===================================================
+
+Symmetric CORN utilizes the concept of Market symmetry which states that markets have mirrored price movements. Increasing price trends
+represents a mirror of a decreasing trend. This gives us an intuitional understanding that if the price movements are strongly negatively
+correlated, the optimal portfolio weights should minimize the returns or the losses from those periods as it is most likely that the optimal
+portfolio weights would be the inverse.
+
+Introduced recently in a Journal of Financial Data Science paper by Yang Wang and Dong Wang in 2019, SCORN identifies positively
+correlated windows and negatively correlated windows.
+
+The positiviely correlated windows are identified similar to the process for CORN.
+
+.. math::
+    C(x_t;w,\rho) = \lbrace x_j \vert R(X^{j-1}_{j-2},X^{t-1}_{t-w})  > \rho)
+
```

Comment:
Should in this and below formula be `R(X^{j-1}_{j-w},X^{t-1}_{t-w})` instead of `R(X^{j-1}_{j-2},X^{t-1}_{t-w})` ?

Reply:
nice catch! yes